tensorflow/contrib/tensorrt/resources/trt_int8_calibrator.cc

```diff
@@ -65,32 +62,32 @@ bool TRTInt8Calibrator::setBatch(
 
     // TODO(aaroey): we should not use sync copy on default stream. Make sure
     // stream->ThenMemcpy() is used in future PRs.
-    auto status =
-        cudaMemcpy(d.first, it.second, d.second, cudaMemcpyDeviceToDevice);
+    auto status = cudaMemcpyAsync(d.first, it.second, d.second,
+                                  cudaMemcpyDeviceToDevice, stream);
     if (status != cudaSuccess) {
       LOG(FATAL) << "cudaMemcpy " << engine_name_ << " for '" << it.first
                  << "' failed with " << status;
     }
   }
-  calib_running_.store(true, std::memory_order_release);  // release builder
+  cudaStreamSynchronize(
```

Comment:
This is expensive, can we avoid the copy? Does the calibration run on a stream different than the one used by the trt engine?

Also if we really want to use sync copy we can use stream->SynchronousMemcpy() instead.